server.go

```diff
@@ -153,6 +153,57 @@ func createPullSecret(clientset *kubernetes.Clientset, ns *corev1.Namespace, cre
 	return nil
 }
 
+// deletePullSecret deletes the image registry pull secret for the provided namespace
+func deletePullSecret(clientset *kubernetes.Clientset, ns corev1.Namespace) error {
+	secrets := clientset.CoreV1().Secrets(ns.Name)
+	if err := secrets.Delete(context.TODO(), gcpAuth, metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("deleting %s secret in %s namespace: %v", gcpAuth, ns.Name, err)
+	}
+	return nil
+}
+
+// refreshAllPullSecrets deletes and recreates image registry pull secrets for all namespaces
+func refreshAllPullSecrets() error {
```

Comment:
lets generate tokens for 12 hours in first place so dont have to refresh it that often

Reply:
I tried but when I try to change the expiry it still gives the same value
```
package main

import (
        "context"
        "fmt"
        "log"
        "time"

        "golang.org/x/oauth2"
        "golang.org/x/oauth2/google"
)

func main() {
        creds, err := google.FindDefaultCredentials(context.Background())
        if err != nil {
                log.Fatal(err)
        }
        token, err := creds.TokenSource.Token()
        if err != nil {
                log.Fatal(err)
        }
        fmt.Println(token.Expiry)
        dur, err := time.ParseDuration("30s")
        if err != nil {
                log.Fatal(err)
        }
        ts := oauth2.ReuseTokenSourceWithExpiry(token, creds.TokenSource, dur)
        token, err = ts.Token()
        if err != nil {
                log.Fatal(err)
        }
        fmt.Println(token.Expiry)
} 
```